src/main/java/com/mongodb/jdbc/MongoSQLDatabaseMetaData.java

```diff
@@ -190,10 +208,142 @@ public ResultSet getClientInfoProperties() throws SQLException {
         throw new SQLFeatureNotSupportedException("TODO");
     }
 
+    private MongoJsonSchema getFunctionJsonSchema() {
+        MongoJsonSchema schema = new MongoJsonSchema();
+        schema.bsonType = "object";
+        schema.additionalProperties = false;
+        schema.required = new HashSet<String>();
+        schema.properties = new HashMap<String, MongoJsonSchema>();
+
+        schema.required.add(FUNCTION_CAT);
+        schema.required.add(FUNCTION_SCHEM);
+        schema.required.add(FUNCTION_NAME);
+        schema.required.add(REMARKS);
+        schema.required.add(FUNCTION_TYPE);
+        schema.required.add(SPECIFIC_NAME);
+
+        schema.properties.put(FUNCTION_CAT, stringSchema);
+        schema.properties.put(FUNCTION_SCHEM, stringSchema);
+        schema.properties.put(FUNCTION_NAME, stringSchema);
+        schema.properties.put(REMARKS, stringSchema);
+        schema.properties.put(FUNCTION_TYPE, intSchema);
+        schema.properties.put(SPECIFIC_NAME, stringSchema);
+        return schema;
+    }
+
+    private BsonDocument getFunctionValuesDoc(String functionName, String remarks) {
```

Comment:
I think that's extra complexity that doesn't really help in this case. Now, if there were some base class functionality that used it, it would make sense, but right now it's only used in the base class. Using the same name is just to make it more clear to someone familiar with the other class.